scikeras/wrappers.py

```diff
@@ -649,32 +634,15 @@ def _fit(self, X, y, sample_weight=None, warm_start=False):
             # because the predictions differ by a small margin).
             # To get around this, we manually delete these samples here
             zeros = sample_weight == 0
+            if zeros.sum() == zeros.size:
+                raise RuntimeError(
+                    "Cannot train because there are no samples"
+                    " left after deleting points with zero sample weight!"
```

Comment:
Shouldn't a ValueError be raised? That's what the documentation on [ValueError][2] and [RuntimeError][1] suggests. An incorrect value (the sample weight) was passed.

[1]:https://docs.python.org/3/library/exceptions.html#RuntimeError
[2]:https://docs.python.org/3/library/exceptions.html#ValueError

Reply:
Also, style nit: this might be a better message:

> No training samples had any weight; only zeros were passed in sample_weight={}. That means there's nothing to train on by definition, so training can not be completed.

Reply:
Sure, I can change it. The thought process behind RuntimeError is that a value of all zeros for sample weight is not technically wrong, but if we delete all samples then that probably causes issues in Keras. I felt that it could be confusing to users if we say that their value for `sample_weight` is invalid if it is normally valid in `sklearn` (even if it is nonsensical).

Reply:
But thinking about it more, I think it's fine to raise a ValueError for nonsensical inputs, even if some sklearn estimators might succeed.